src/data/common/Dhis2DataFormRepository.ts

```diff
@@ -84,84 +84,97 @@ export class Dhis2DataFormRepository implements DataFormRepository {
-                                : undefined,
-                            rules: dataElementRules,
-                            htmlText: deConfig?.texts?.name,
+                            viewType: config.viewType,
+                            calculateTotals: config.calculateTotals,
+                            subNationals: config?.subNationalDataset
+                                ? _(configDataForm.subNationals)
+                                      .filter((sn: SubNational) => sn.parentId === orgUnit)
+                                      .sortBy(sn => sn.name)
+                                      .value()
+                                : [],
+                            ...base2,
                         };
-                    })
-                    .compact()
-                    .value(),
-                titleVariant: config?.titleVariant,
-                styles: SectionStyle.buildSectionStyles(config?.styles),
-                columnsDescriptions: config?.columnsDescriptions,
-                groupDescriptions: config?.groupDescriptions,
-                totals: config?.totals,
-                showRowTotals: section.showRowTotals,
-                toggleMultiple: config?.toggleMultiple ? buildToggleMultiple(config.toggleMultiple, dataElements) : [],
-            };
 
-            if (!config) return { viewType: "table", calculateTotals: undefined, ...base };
-
-            const base2 = getSectionBaseWithToggle(config, base, dataElements);
-
-            switch (config.viewType) {
-                case "grid-with-periods":
-                    return { viewType: config.viewType, periods: config.periods, ...base2 };
-                case "table":
-                case "grid":
-                case "grid-with-totals":
-                    return { viewType: config.viewType, calculateTotals: config.calculateTotals, ...base2 };
-                case "grid-with-subnational-ous":
-                    return {
-                        viewType: config.viewType,
-                        calculateTotals: config.calculateTotals,
-                        subNationals: config?.subNationalDataset
-                            ? _(configDataForm.subNationals)
-                                  .filter((sn: SubNational) => sn.parentId === orgUnit)
-                                  .sortBy(sn => sn.name)
-                                  .value()
-                            : [],
-                        ...base2,
-                    };
-
-                default:
-                    return { viewType: config.viewType, ...base2 };
-            }
-        });
+                    default:
+                        return { viewType: config.viewType, ...base2 };
+                }
+            })
+            .sortBy([
+                section => (section.tabs.order ? parseInt(section.tabs.order?.split(".")[0] || "0", 10) : 0),
+                section => (section.tabs.order ? parseInt(section.tabs.order?.split(".")[1] || "0", 10) : 0),
```

Comment:
The second parameter in the parseInt method is already 10 for decimal numbers so for this case I think it's safe to remove it.